AirCasting/Graph/External/ExternalGraph.swift

```diff
@@ -0,0 +1,168 @@
+import SwiftUI
+import Charts
+import Resolver
+
+struct ExternalGraph: UIViewRepresentable {
+    typealias UIViewType = AirCastingGraph
+    typealias OnChange = (ClosedRange<Date>) -> Void
+    
+    @InjectedObject private var userSettings: UserSettings
+    @ObservedObject var stream: MeasurementStreamEntity
+    @ObservedObject var thresholds: SensorThreshold
+    private var rangeChangeAction: OnChange?
+    
+    var isAutozoomEnabled: Bool
+    let simplifiedGraphEntryThreshold = 1000
+    
+    init(stream: MeasurementStreamEntity, thresholds: SensorThreshold, isAutozoomEnabled: Bool) {
```

Comment:
I'm asking out of curiosity: since this view doesn't get sessionEntity as an argument, why did you create a new view for external sessions? MeasurementsStreams are the same for both types of sessions.

Reply:
Sure, but I think that having the same instance of Graph for two things (external and internal sessions) can give us less possibility for flexible modification in the future. I mean by that; the external session is mostly the same in our app, but at the same time way different then internal one -  so splitting it into different ones can help us in the future I believe

Reply:
I cannot agree here with you. If we could have some views done in a more generic way then we could always apply some initial version of it and after that extend it according to the needs (of let's say) current implementation. What I mean by that is: having an ex. `CoreGraphView` and then inheriting from it in ex `ExternalGraphView` and `InternalGraphView` would allow us to explicitly state what is common and change it if the future would not affect anything else (possibly). In our case, the internal graph has notes when the second does not (maybe also it has some other changes which I do not remember). Having one graph here means always the same in both places.

At least for me, it is the same with having different structs for APIs or data which we are saving. We tend to make a few structs for the same data (even if those structs are sometimes exactly the same) based on the "file" so that in the future we could protect ourselves from changing most of the code; the change only would be done in the file where we really need it (most of the time, because passing data and so on comes here, but most of the time...)

Reply:
Thanks for a detailed response. Sure, I forgot about the notes. :) In that case, I think it is good to keep them separately. But as you mention, it would be cool to have a parent class with all of the graph stylings.